### PR TITLE
Hotfix subscription crash for failing token provider

### PR DESCRIPTION
**Pull request by marekoid**
Issue reported by a user (internal ticket id: 19417).

### CODE REVIEW

**Comment by marekoid** [comment]
Snyk is not happy but that some generic config issue, nothing to do with this PR (as present also on others), hence the PR is ready for the review, enjoy :)

File: pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/ResumingSubscription.kt [hunk 1]
```diff
@@ -38,7 +38,15 @@ private class ResumingSubscription<A>(
 ) : Subscription {
 
     private val onTransition: StateTransition = { newState ->
-        logger.verbose("ResumingSubscription $subscriptionID: transitioning to ${newState.javaClass.simpleName} (from ${state.javaClass.simpleName})")
+        // Safe calls are necessary as there's an edge case initialisation cycle
+        // that leads to this being called back when state is being created (still is null).
+        // Breaking initialisation and performing actions will be addressed later.
+        // For now this is a safe workaround for the issue (#19417)
```
Comment (mdpye) [inline pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/ResumingSubscription.kt]:
```suggestion
        // For now this is a safe workaround for the issue (zendesk #19417)
```

**Comment by mdpye** [comment]
@marekoid can we roll #49 in to the same release please?

File: pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/ResumingSubscription.kt (hunk 1 above)
Comment (marekoid) [reply pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/ResumingSubscription.kt]: [Resolved](https://github.com/pusher/pusher-platform-android/pull/50/commits/0aa13d8c2ae6c37f09ce23b66c3151202089fa68) (with uppercase as besides the branding logo, the product name is being written like that on their site).

**Pull request merged**